equipmanager.lic

```diff
@@ -184,37 +184,51 @@ class EquipmentManager
     return false
   end
 
+  # Unlike `wield_weapon`, does NOT stow what's in your hands (not sure if that's on purpose or not).
+  # Handles swapping the weapon to desired weapon skill (e.g. bastard swords, bar maces, ristes).
+  # Will ensure the item ends up in your left hand.
   def wield_weapon_offhand(description, skill = nil)
     return if description.empty?
 
     weapon = item_by_desc(description)
     unless weapon
-      echo("failed to match a weapon for #{description}")
-      return
+      DRC.message("Failed to match a weapon for #{description}:#{skill}")
+      return false
+    end
+
+    if get_item?(weapon)
+      swap_to_skill?(weapon.name, skill) if skill && weapon.swappable
+      DRC.bput('swap', 'You move') if DRCI.in_right_hand?(weapon)
+      return true
     end
 
-    get_item?(weapon)
-    swap_to_skill(weapon.name, skill) if skill && weapon.swappable
+    return false
   end
 
+  # Stows any items in your hands then wields the requested item to your right hand.
+  # Handles swapping the weapon to desired weapon skill (e.g. bastard swords, bar maces, ristes).
+  # If the skill to swap to is 'Offhand Weapon' then will swap it to your left hand.
   def wield_weapon(description, skill = nil)
     return if description.empty?
 
     offhand = skill == 'Offhand Weapon'
     weapon = item_by_desc(description)
     unless weapon
-      echo("failed to match a weapon for #{description}:#{skill}")
-      return
+      DRC.message("Failed to match a weapon for #{description}:#{skill}")
+      return false
     end
 
     if [left_hand, right_hand].grep(weapon.short_regex).any?
       stow_weapon
     end
 
-    get_item?(weapon)
+    if get_item?(weapon)
+      swap_to_skill?(weapon.name, skill) if skill && weapon.swappable
+      DRC.bput('swap', 'You move') if offhand && right_hand
+      return true
```

Comment:
These are now indented within `get_item?` so that they are only tried if the weapon is indeed in your hands